eth/fetcher/block_fetcher.go

```diff
@@ -705,6 +710,14 @@ func (f *BlockFetcher) enqueue(peer string, header *types.Header, block *types.B
 // the phase states accordingly.
 func (f *BlockFetcher) ImportBlocks(peer string, block *types.Block, relay bool) {
 	hash := block.Hash()
+	nodeCtx := common.NodeLocation.Context()
+
+	// Check if the Block is atleast half the current difficulty in Zone Context,
+	// this makes sure that the nodes don't listen to the forks with less than
+	// 50% of current difficulty
+	if nodeCtx == common.ZONE_CTX && block.Difficulty().Cmp(new(big.Int).Div(f.currentDifficulty(), big.NewInt(2))) < 0 {
```

Comment:
We need to check hash, not difficulty. Anyone can put any arbitrary number in the difficulty field. It would be hard to come up with a hash that was half the work as our target.

Reply:
This is just for a noisy block filtering, I think both Hash or Difficulty will work, using your logic, if you can modify difficulty hash is also modified. Its not a problem if someone lowers the difficulty.

Reply:
No, a hash cannot be modified. We compute the hash, and it is not possible for them to find a block with a low hash without putting in real work to do so. Hence proof of work.

We talked about this feature before. This is to prevent the node from getting overwhelmed by low-work spam blocks, and the only way to achieve this is by checking work, not difficulty.